src/app/modules/myinfo/myinfo.controller.ts

```diff
@@ -0,0 +1,206 @@
+import { celebrate, Joi, Segments } from 'celebrate'
+import { Request } from 'express'
+import { Query, RequestHandler } from 'express-serve-static-core'
+import { StatusCodes } from 'http-status-codes'
+
+import { createLoggerWithLabel } from '../../../config/logger'
+import { AuthType } from '../../../types'
+import { createReqMeta } from '../../utils/request'
+import * as FormService from '../form/form.service'
+import { SpcpFactory } from '../spcp/spcp.factory'
+import { LoginPageValidationResult } from '../spcp/spcp.types'
+
+import { MYINFO_COOKIE_NAME, MYINFO_COOKIE_OPTIONS } from './myinfo.constants'
+import { MyInfoFactory } from './myinfo.factory'
+import { MyInfoCookiePayload, MyInfoCookieState } from './myinfo.types'
+import {
+  mapEServiceIdCheckError,
+  mapRedirectURLError,
+  validateMyInfoForm,
+} from './myinfo.util'
+
+const logger = createLoggerWithLabel(module)
+
+const validateRedirectURLRequest = celebrate({
+  [Segments.QUERY]: {
+    formId: Joi.string()
+      .regex(/^[0-9a-fA-F]{24}$/)
+      .required(),
+    rememberMe: Joi.boolean().required(),
```

Comment:
MyInfo logins shouldn't provide the "Remember me" checkbox

Reply:
unless... this is part of the plan to have Singpass and MyInfo share the same redirect URL endpoint for the time being?

Reply:
nope, was planning to remove this! I initially added it so it would be easier to support longer MyInfo sessions in future, but now I think it's better to leave it out because we don't know what MyInfo's future implementation of extended sessions will look like